src/containers/Login/index.jsx

```diff
@@ -0,0 +1,288 @@
+
+const INITIAL_UPDATE_PASSWORD = {
+  value: "",
+  updatePasswordError: "",
+};
+
+function Login() {
+  const { isLoggedIn, isPasswordUpdate, loading, error, name } = useSelector(
+    (state) => state.user
+  );
+  const [userData, setUserData] = useState(INITIAL_STATE);
+  const [errorData, setErrorData] = useState(INITIAL_ERROR_DATA);
+  const [updatePasswordObj, setUpdatePasswordObj] = useState(
+    INITIAL_UPDATE_PASSWORD
+  );
+
+  const dispatch = useDispatch();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isLoggedIn && !isPasswordUpdate) {
+      navigate(LINKS.home);
+    }
+  }, [isLoggedIn, isPasswordUpdate]);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setUserData({ ...userData, [name]: value });
+  };
+
+  const handleChangeUpdatePassword = (event) => {
+    let value = event.target.value;
+    setUpdatePasswordObj((obj) => ({ ...obj, value }));
+  };
+
+  const loginUser = (event) => {
+    event.preventDefault();
+    setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    console.log("clicked login");
+    const errorResult = validation(userData);
+
+    if (errorResult) {
+      setErrorData((obj) => ({ ...obj, ...errorResult }));
+      return;
+    } else {
+      //dispatch login action
+      //submit user data
+      dispatch(login({ ...userData, appType: "music" }));
+      console.log(userData);
+      setUserData((obj) => ({ ...obj, ...INITIAL_STATE }));
+      setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    }
+  };
+
+  const updatePasswordFunction = (event) => {
+    event.preventDefault();
+    setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    console.log("clicked login");
+    const errorResult = validation(userData);
+
+    if (errorResult) {
+      setErrorData((obj) => ({ ...obj, ...errorResult }));
+      return;
+    } else {
+      //dispatch login action
+      //submit user data
+      const submitData = {
+        name,
+        email: userData.email,
+        passwordCurrent: userData.password,
+        password: updatePasswordObj.value,
+      };
+      dispatch(updatePassword({ ...submitData, appType: "music" }));
+      console.log(submitData);
+      setUserData((obj) => ({ ...obj, ...INITIAL_STATE }));
+      setErrorData((obj) => ({ ...obj, ...INITIAL_ERROR_DATA }));
+    }
+  };
+
+  const validation = (userData) => {
+    const { email, password } = userData;
+
+    if (!email || !password) {
+      return {
+        emailError: "Please enter all fields",
+        passwordError: "Please enter all fields",
+      };
+    }
+    let emailError =
+      email.length > 3 && email.includes("") ? "" : "Please enter valid email";
+    let passwordError =
+      password.length > 3 ? "" : "password must contain atleast 3 letters";
+    console.log(emailError, passwordError);
+
+    if (!emailError && !passwordError) return false;
+    return {
+      emailError,
+      passwordError,
+    };
+  };
```

Comment:
same validation logic used in signup as well please move this function to util and use from there whereever you have used the same validation logic